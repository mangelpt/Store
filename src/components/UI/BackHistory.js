import React from 'react'
import styled from '@emotion/styled';

const StyledDiv = styled.div`
width: 75vw;
display: flex;
height: fit-content;
font-size: 20px;
font-weight: 600;
margin:10px auto;
padding: 10px;
gap: 70px;
button{
  border: none;
  background-color: transparent;
  cursor: pointer;
}
svg{
  width: 20px;
  color: #000;
}
`;
export const BackHistory = ({ justify, children }) => {
  return (
    <StyledDiv style={{ justifyContent: `${justify}` }} >
      {children}
    </StyledDiv>
  )
}
