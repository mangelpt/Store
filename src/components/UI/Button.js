import styled from "@emotion/styled"

const Bttn = styled.button`
  background: #FA4A0C;
  border-radius: 30px;
  padding: 12px 16px;
  height: 70px;
  width: 310px;
  font-weight: 600;
  font-size: 18px;
  line-height: 23px;
  color: #FFFFFF;
  text-align: center;
  border: none;
  cursor: pointer;
  
  ${props => props.prefix==="foot" && `
    position: fixed;
    bottom: 104px;
  `} 

  &:focus {
    outline: none;
  }
`

export function Button(props) {
  return (
    <Bttn onClick={props.fnc} prefix={props.prefix}>{props.text}</Bttn>
  )
}
