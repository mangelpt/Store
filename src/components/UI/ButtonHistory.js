import styled from '@emotion/styled';
import { Link } from "react-router-dom";
import { ArrowIconRight } from "./Icons";

const StyledDiv = styled.div`
.link{
  padding: 15px 15px;
  text-decoration: none;
  font-size: 18px;
  font-weight: bold;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  border: none;
  border-radius: 20px;
  cursor: pointer;
  margin: auto;
  width: 75vw;
  color: #000;
  svg {
    width: 25px;
  }
}

`;
export const ButtonHistory = () => {
  return (
    <StyledDiv>
      <Link to="/history" className="link">
        History
        <ArrowIconRight />
      </Link>
    </StyledDiv>
  )
}

