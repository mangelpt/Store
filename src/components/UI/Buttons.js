import styled from "@emotion/styled";

export const ButtonWhite = styled.button`
  padding: 15px 15px;
  font-size: 18px;
  font-weight: bold;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: orange;
  border: none;
  border-radius: 20px;
  cursor: pointer;
  margin: auto;
  width: 75vw;
  color: #000;
  svg {
    width: 25px;
  }
`;
