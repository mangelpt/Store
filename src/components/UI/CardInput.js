import styled from "@emotion/styled"
import { Lbl } from "./Labels"
import  {Input} from "./Input"

const ContainerInput = styled.div`
    display: flex;
    flex-direction: column;
    width:208px;
    height:50px;
`

export function CardInput({textlabel, name, type, placeholder}) {
  return (
    <>
    <ContainerInput>
      <Lbl>{textlabel}</Lbl>
      <Input type={type}
          name={name}
          placeholder={placeholder}
      >
      </Input>
    </ContainerInput>  
    </>
  )
}