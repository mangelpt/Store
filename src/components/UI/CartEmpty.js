import styled from "@emotion/styled";

export function CartEmpty() {

  const ContainerCartEmpty = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    svg {
      width: 100px;
      height: 100px;
    }
    .iconcart {
      fill: none;
      stroke: #C7C7C7;
    }
    h1 {
      font-family: "Source Sans Pro";
      font-style: normal;
      font-weight: 600;
      font-size: 28px;
      line-height: 35px;
      text-align: center;
      color: #333333;
    }
  `;

  return (
    <ContainerCartEmpty>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        class="h-6 w-6"
        fill="none"
        viewBox="0 0 24 24"
        stroke="currentColor"
      >
        <path
          class="iconcart"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
        />
      </svg>
      <h1>No order yet</h1>
    </ContainerCartEmpty>
  );
}
