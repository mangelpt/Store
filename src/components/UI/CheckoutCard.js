import styled from "@emotion/styled";

const Checkout = styled.div`
  background: #FFFFFF;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.03);
  border-radius: 20px;
  padding: 25px 53px 20px 30px;
  width: 232px;
  margin-bottom: 10px;
  form {
    width: 232px;
    display: flex;
    flex-direction: column;
  }

  input {
    font-weight: 400;
    font-size: 18px;
    line-height: 23px;
    color: #333333;
    border: none;
    background: none;
    outline: none;
  }

  input[name="name"] {
    font-weight: 600;
  }

  input::placeholder {
    opacity: 0.5;
  }

  div {
    margin: 7px 0 10px 0;
    opacity: 0.2;
    border-top: 0.5px solid #000000; 
  }
`;

export function CheckoutCard(props) {
  return (
    <Checkout>
      <form>
        <input
          type="text"
          name="name"
          value={props.name}
          placeholder="name"
          onChange={props.onChange}
          disabled={props.disabled}
        />
        <div />
        <input
          type="text"
          name="address"
          value={props.address}
          onChange={props.onChange}
          placeholder="address" 
          disabled={props.disabled}
        />
        <div />
        <input
          type="tel"
          name="phone"
          value={props.phone}
          placeholder="phone"
          onChange={props.onChange}
          disabled={props.disabled}
        />
      </form>
    </Checkout>
  )
}