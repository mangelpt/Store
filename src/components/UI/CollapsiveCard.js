import styled from "@emotion/styled";
import { useState } from "react";
import { ArrowIconDown, ArrowIconUp } from "./Icons";

export const Collapse = ({ date, amount, total, products, address }) => {
  const [isCollapsed, setIsCollapsed] = useState(false);

  return (
    <Collapsed>
      <div className="content">
        <p>{date}</p>
        <div>
          <span>{`${amount} items`}</span>
          <span>{`$${total}`}</span>
        </div>
      </div>

      <div
        className={`collapse-content ${isCollapsed ? "expanded" : "collapsed"}`}
        aria-expanded={isCollapsed}
      >
        <div className="Order">
          <h3>Order</h3>
          <div>
            <li>
              <span> 1- veggie tomato mix </span> <span>{`$${15.45}`}</span>
            </li>
            <li>
              <span> 1- veggie tomato mix </span> <span>{`$${15.45}`}</span>
            </li>
          </div>
        </div>
        <div className="Delivery">
          <h3>Delivery</h3>
          <span>{address}</span>
        </div>
      </div>

      <button onClick={() => setIsCollapsed(!isCollapsed)}>
        {isCollapsed ? <ArrowIconUp /> : <ArrowIconDown />}
      </button>
    </Collapsed>
  );
};

const Collapsed = styled.div`
  width: 75vw;
  background-color: #fff;
  border-radius: 10px;
  margin: 10px auto;
  display: flex;
  flex-direction: column;
  padding: 10px;
  .content {
    display: flex;
    flex-direction: column;
    font-size: 18px;
    gap: 10px;
    color: gray;
    div {
      background-color: #fff;
      display: flex;
      justify-content: space-between;
      font-size: 16px;
      & :last-child {
        color: orange;
        font-weight: bold;
      }
    }
  }
  .Order {
    list-style: none;
    margin-top: 15px;
    margin-bottom: 10px;
    div {
      border-bottom: 0.5px solid gray;
      padding-bottom: 20px;
      li {
        color: gray;
        display: flex;
        justify-content: space-between;
        font-size: 16px;
      }
    }
  }
  .Delivery {
    padding: 15px 2px;
    span {
      color: gray;
    }
  }
  button {
    margin-top: 5px;
    cursor: pointer;
    align-self: flex-end;
    background-color: transparent;
    border: none;
    svg {
      width: 20px;
    }
  }
  .collapse-content.collapsed {
    display: none;
  }
  .collapsed-content.expanded {
    display: flex;
  }
  .text-format {
    text-align: left;
    font-size: 14px;
    line-height: 17px;
  }
`;
