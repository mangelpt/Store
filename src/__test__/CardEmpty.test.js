import React from "react";
import renderer from "react-test-renderer";
import { CartEmpty } from "../components/UI/CartEmpty";

describe("Styled Label", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CartEmpty/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});