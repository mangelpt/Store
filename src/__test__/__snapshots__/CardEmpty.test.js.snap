// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Label renders correctly 1`] = `
<div
  className="css-16j4y09"
>
  <svg
    className="h-6 w-6"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
    />
  </svg>
  <h1>
    No order yet
  </h1>
</div>
`;
